### ISSUE TITLE
Watch and reload api folder

### ISSUE THREAD

**Issue by pistacchio** [nuxt/nuxt.js#1819]
How to make nuxt watch for "non standard" directories and recompile / reload itself, and more specifically for dirs with additional server apis?

I have my express api in `~/api/`. Since I reference the directory in `serverMiddleware` with '~/api', I would expect Nuxt to reload when I make some changes to the files in that dir, but it doesn't.

You can reproduce it easily with the "auth routes" example. Any change in the `api` folder doesn't trigger the rebuild action. Since the initial build is not exactly fast, having to restop / restart the server manually for every change in the site api can become boring easily.

I'm simply using npm run dev that fires nuxt, I don't have any direct control on either **nodemon** (if it's used internally) or on **webpack** (that I'm pretty sure is). I tried adding `watch: [ '~/api/*.js']`, `watch: [ '~/api/index.js']`, `watch: [ '~/api/**/*.js']` to build in `nuxt.conf.js` but with no luck.

**Comment by stursby** [comment]
@pistacchio So I tried the [auth routes example](https://github.com/nuxt/nuxt.js/tree/dev/examples/auth-routes) and added the following to `nuxt.config.js`:

```js
module.exports = {
  build: {
    watch: ['api']
  }
}
```
Now when saving `/api/index.js` I can see nuxt reload in the Terminal logs, however, since it's running an express server... the express server still needs to be reloaded to reflect the changes.

**Comment by pistacchio** [comment]
This has helped, thank you very much )

**Comment by AlexRadch** [comment]
I installed `nodemon` (`npm install --save-dev nodemon`)

Then I changed `package.json` 
```javascript
{
  "scripts": {
    "dev": "nodemon --watch api --exec \"nuxt\""
  }
}

**Comment by northamerican** [comment]
thanks for the above, however it doesn't reload the page as it would with a vue file. nuxt makes use of `webpack-hot-middleware` for hot reloading, maybe this can be used?

**Comment by kiwicopple** [comment]
Looks like this is solved by https://github.com/nuxt/nuxt.js/issues/1509

You'll need to use `nuxt-edge`

**Comment by galvez** [comment]
@kiwicopple Nuxt 2 has already been released :)

**Comment by kiwicopple** [comment]
sorry! new to Nuxt and I was using boilerplates which were on 1.0

Someone may want to inform [all ](https://medium.com/@chanlitohem/march-29th-and-now-were-in-almost-august-still-nuxt-2-hasn-t-been-officially-released-and-lots-2320c87c68fc) [these ](https://medium.com/@cong.nguyen/hi-e00d7cb7b3fd) [people ](https://medium.com/@dmikester1/were-sneaking-up-on-september-now-any-updates-on-2-0-thanks-for-all-your-work-ca6a8ad47b65) in the comments too

**Comment by manniL** [comment]
See #4301
